--- libsap-1.51.1.orig/Makefile	Mon Jan  8 00:02:01 2001
+++ libsap-1.51.1/Makefile	Tue Feb  5 22:55:50 2002
@@ -1,4 +1,4 @@
-OBJECTS= sapCpu.o sapPokey.o pokey0.o pokey1.o sapEngine.o
+OBJECTS= sapCpu.lo sapPokey.lo pokey0.lo pokey1.lo sapEngine.lo
 OPTS=-O2 
 AR=ar
 
@@ -16,22 +16,18 @@
 
 dynamic: libsap.so
 
-$(OBJECTS): %.o: %.cpp
-	$(CC) $(OPTS) -o $@ -c $<
+$(OBJECTS): %.lo: %.cpp
+	libtool --mode=compile $(CC) $(OPTS) -o $@ -c $<
 
-libsap.a: $(OBJECTS)
-	$(AR) rc $@ $(OBJECTS)
-
-libsap.so: $(OBJECTS)
-	$(CC) $(OBJECTS) -shared -o $@
+libsap.la: $(OBJECTS)
+	libtool --mode=link $(CC) $(OBJECTS) -o $@ -rpath $(libdir) -version-info 1:51
 
 clean:
 	rm -f $(OBJECTS) libsap.so libsap.a
 
 install:
-	install -Dm644 libsap.h /usr/local/include/libsap.h
-	if [ -f libsap.a ]; then install -Dm755 libsap.a /usr/local/lib/libsap.a; fi
-	if [ -f libsap.so ]; then install -Dm755 libsap.so /usr/local/lib/libsap.so; fi
+	install -Dm644 libsap.h $(DESTDIR)/usr/include/libsap.h
+	libtool --mode=install install -D libsap.la $(DESTDIR)$(libdir)/libsap.la
 
 uninstall:
 	rm -f /usr/local/include/libsap.h /usr/local/lib/libsap.a /usr/local/lib/libsap.so
